src/overlays/actors/ovl_En_Mb/z_en_mb.c

```diff
@@ -21,99 +20,1417 @@ const ActorInit En_Mb_InitVars = {
+typedef struct {
+    /* 0x0 */ s16 unk_0;
+    /* 0x2 */ s16 unk_2;
+    /* 0x4 */ u16 unk_4;
+} struct_80AA9D68;
 
+struct_80AA9D68 D_80AA9D68 = {
+    0x0014,
+    0x0028,
+    0x0000,
+};
+
+typedef struct {
+    s16 unk_0;
+    s16 unk_2;
+    s16 unk_4;
+} struct_80AA9D70;
+
+struct_80AA9D70 D_80AA9D70 = {
+    0xF63C,
+    0x0000,
+    0x0DAC,
+};
+
+// Weird Data issue
 #pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Mb/func_80AA7938.s")
 
+#ifdef NON_MATCHING
+// Small stack
+void func_80AA7CAC(EnMb* this, GlobalContext* globalCtx) {
+    Player* player;
+    s32 currentFrame;
+    s32 sp4C;
+    s32 sp48;
+
+    f32 sp40;
+
+    player = PLAYER;
+    sp4C = 0;
+    sp48 = !func_800339B8(&this->actor, globalCtx, 110.0f, this->actor.posRot.rot.y);
+    currentFrame = (s32)this->skelAnime.animCurrentFrame;
+    if (SkelAnime_FrameUpdateMatrix(&this->skelAnime) != 0) {
+        SkelAnime_ChangeAnimDefaultRepeat(&this->skelAnime, &D_06002F10);
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_MORIBLIN_ATTACK);
+    }
+    if (this->unk_32E != 0) {
+        this->unk_32E--;
+        Math_SmoothScaleMaxMinS(&this->actor.shape.rot.y, this->actor.posRot.rot.y, 1, 0x1F40, 0);
+        sp48 = 0;
+    } else {
+        this->actor.speedXZ = 10.0f;
+        this->attackParams = 1;
+        func_80033260(globalCtx, &this->actor, &this->actor.posRot.pos, 5.0f, 3, 4.0f, 0x64, 0xF, 0);
+        if (currentFrame != (s32)this->skelAnime.animCurrentFrame) {
+            if (((s32)this->skelAnime.animCurrentFrame == 2 || (s32)this->skelAnime.animCurrentFrame == 6)) {
+                Audio_PlayActorSound2(&this->actor, NA_SE_EN_MORIBLIN_DASH);
+            }
+        }
+    }
+    if ((this->collider2.base.atFlags & 2)) {
+        if (&player->actor == this->collider2.base.at) {
+            if ((sp48 == 0) && ((player->stateFlags2 & 0x80) == 0)) {
+                if (player->invincibilityTimer < 0) {
+                    if (player->invincibilityTimer < -0x27) {
+                        player->invincibilityTimer = 0;
+                    } else {
+                        player->invincibilityTimer = 0;
+                        globalCtx->damagePlayer(globalCtx, -8);
+                    }
+                }
+                if ((this->collider2.base.atFlags & 4) == 0) {
+                    Audio_PlayActorSound2(&player->actor, NA_SE_PL_BODY_HIT);
+                }
+                if (globalCtx->grabPlayer(globalCtx, player) != 0) {
+                    player->actor.parent = &this->actor;
+                }
+            }
+            sp4C = 1;
+        } else {
+            this->collider2.base.atFlags &= -3;
+        }
+    }
+    if (((player->stateFlags2 & 0x80) != 0) && (&this->actor == player->actor.parent)) {
+        sp40 = Math_Coss(this->actor.shape.rot.y);
+        player->actor.posRot.pos.x =
+            ((Math_Sins(this->actor.shape.rot.y) * 89.0f) + (this->actor.posRot.pos.x + (sp40 * 10.0f)));
```

Comment:
```suggestion
        player->actor.posRot.pos.x = this->actor.posRot.pos.x + (Math_Coss(this->actor.shape.rot.y) * 10.0f) + ((Math_Sins(this->actor.shape.rot.y) * 89.0f) ;
```
The temp is unnecessary. Same applies below.